dali/c_api/c_api_test.cc

```diff
@@ -69,9 +69,27 @@ std::string GetDeviceStr(device_type_t dev) {
   return dev == CPU ? "cpu" : "gpu";
 }
 
+// Allocates a buffer on the specified backend and, if necessary, another one for the CPU
```

Comment:
The only test we run from this file without GPU is CpuOnlyTest - see https://github.com/NVIDIA/DALI/blob/main/qa/TL0_cpu_only/test.sh#L34.
Maybe we should improve the pattern there - or change test naming here to run more CPU only test?